src/shared/dns-domain.c

```diff
@@ -1324,3 +1324,15 @@ int dns_name_apply_idna(const char *name, char **ret) {
 
         return (int) n;
 }
+
+int dns_name_is_valid_or_address(const char *name) {
+        /* Returns > 1 if the specified name is either a valid IP address formatted as string or a valid DNS name */
```

Comment:
well, dns_name_is_valid() might return an error (since it can fail due to OOM), and I think we shouldn't eat that up here...